File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l, nums = 0, set()
        max_length = 0
        for r, char in enumerate(s):
            while char in nums:
                nums.remove(s[l])
                l += 1
            nums.add(char)
            max_length = max(max_length, len(nums))
            print(nums, max_length)
        return max_length

File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-2.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int l = 0, max_ones = 0;
        for (int r = 0; r < ssize(nums); r++) {
            if (nums[r] == 0) k--;

            while (k < 0) {
                if (nums[l] == 0) k++;
                l++;
            }
            max_ones = max(max_ones, r - l + 1);
        }
        return max_ones;
    }
};